text/0004-api-history-schema.md

```diff
@@ -0,0 +1,216 @@
+    }
+  },
+  "type": "object",
+  "properties": {
+    "added": { "type": "array", "items": { "$ref": "#/definitions/changeSchema" } },
+    "deprecated": { "type": "array", "items": { "$ref": "#/definitions/changeSchema" } },
+    "removed": { "type": "array", "items": { "$ref": "#/definitions/changeSchema" } },
+    "changes": { "type": "array", "items": { "$ref": "#/definitions/changeSchema" } }
+  },
+  "additionalProperties": false
+}
+```
+
+The developer will include information about when/where/how/why an API was:
+
+- Added
+- Changed
+- Deprecated
+- Removed
+
+Along with this, they will include a link to the PR where this change was made
+and preferably include a short description of the change. If applicable, they
+will also include the header for that change from the
+[breaking changes documentation.](https://github.com/electron/electron/blob/main/docs/breaking-changes.md)
+
+The description is limited to a short length for efficiency, if the user wants
+to know more they can look at the included PR.
+
+## Drawbacks
+
+- Adhering to a strict schema limits flexibility. Maybe the developer wants to
+include additional properties for a change.
+
+## Rationale and alternatives
+
+- I believe this schema is simple and features everything the user/developer
+would need.
+- The website will use the PR URLs to determine and display Electron version
+numbers for the changes. This is more efficient than including the version
+numbers in the Markdown docs themselves. It also removes the need to manually
+update the docs when a backport is made.
+- Using [Zod](https://zod.dev/) validation logic may be simpler. It's also easy
+to understand for a developer even if they've never worked with it before:
+
+  ```ts
+  const changeSchema = z.object({
+    'pr-url': z.string().url().startsWith('https://github.com/electron/electron/pull/'),
+    'breaking-changes-header': z.string().min(3).optional(),
+    description: z.string().min(3).max(72).optional(),
+  });
+
+  const historySchema = z.object({
+    added: z.array(changeSchema).optional(),
+    deprecated: z.array(changeSchema).optional(),
+    removed: z.array(changeSchema).optional(),
+    changes: z.array(changeSchema).optional(),
+  });
+  ```
+
+## Prior art
+
+I couldn't find an example of something exactly like this being implemented
+elsewhere.
+
+[Semantic Versioning](https://semver.org/) and
+[Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) are
+tangentially related to this concept and both *suggest* a consistent format
+for information about changes but not necessarily *enforce* it.
+
+Of course, plenty of documentation includes versioning.
+
+I think enforcing a schema on API change documentation will massively improve
+the experience of a user upgrading major versions of Electron in their project.
+The process should ideally be as effortless as possible.
+
+## Unresolved questions
+
+- What should be allowed in the descriptions?
```

Comment:
Note from sync: full MD support is probably easiest to implement and we can always check for weird formatting during `e/e` PR review.